From: Markus Elfring <elfring@users.sourceforge.net>
Date: Wed, 4 Feb 2015 11:28:43 +0100
Subject: [PATCH 191/230] cxgb4: Delete an unnecessary check before the
 function call "release_firmware"
Patch-mainline: v4.0-rc1
Git-commit: 0b5b6beea1a3f2b75e5b5baf9df073bf709f9297
References: bsc#924378 FATE#318565 bsc#921338

The release_firmware() function tests whether its argument is NULL and then
returns immediately. Thus the test around the call is not needed.

This issue was detected by using the Coccinelle software.

Signed-off-by: Markus Elfring <elfring@users.sourceforge.net>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: David Chang <dchang@suse.com>
---
 drivers/net/ethernet/chelsio/cxgb4/cxgb4_main.c |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

--- a/drivers/net/ethernet/chelsio/cxgb4/cxgb4_main.c
+++ b/drivers/net/ethernet/chelsio/cxgb4/cxgb4_main.c
@@ -5128,8 +5128,7 @@ static int adap_init0(struct adapter *ad
 				 state, &reset);
 
 		/* Cleaning up */
-		if (fw != NULL)
-			release_firmware(fw);
+		release_firmware(fw);
 		t4_free_mem(card_fw);
 
 		if (ret < 0)
